### PR TITLE
Add support for DROP COLUMN to parser

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: docs/sql/statements/alter-table.rst [hunk 1]
```diff
@@ -173,6 +174,20 @@ expression <sql-literal-value>`.
 
 It's possible to add multiple columns at once.
 
+.. _sql-alter-table-drop-column:
+
+``DROP COLUMN``
+--------------
+
+Can be used to drop a additional column from a table.
```
Comment (BaurzhanSakhariev) [inline docs/sql/statements/alter-table.rst:182]: why additional? maybe just "drop a column"?

File: libs/sql-parser/src/main/java/io/crate/sql/tree/DropColumnDefinition.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    public <U> DropColumnDefinition<U> map(Function<? super T, ? extends U> mapper) {
+        return new DropColumnDefinition<>(mapper.apply(name), ifExists);
+    }
+
+    @Override
+    public <U> TableElement<U> mapExpressions(TableElement<U> mappedElement,
+                                              Function<? super T, ? extends U> mapper) {
+        DropColumnDefinition<U> mappedAddDefinition = (DropColumnDefinition<U>) mappedElement;
+        return new DropColumnDefinition<>(mappedAddDefinition.name, ifExists);
+    }
```
Comment (BaurzhanSakhariev) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/DropColumnDefinition.java:83]: Looks like we are cloning, I think we can drop this method and use default implementation which returns as is. 

We anyway don't have generated expression in `DROP COLUMN` statements

File: libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+package io.crate.sql.tree;
+
+import java.util.List;
+import java.util.Objects;
+
+public class AlterTableDropColumn<T> extends Statement {
+
+    private final Table<T> table;
+    private final List<AddColumnDefinition<T>> addColumnDefinitions;
+
```
Comment (BaurzhanSakhariev) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java:31]: Should be `List<DropColumnDefinition>`

File: libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    public AlterTableDropColumn(Table<T> table, List<AddColumnDefinition<T>> addColumnDefinitions) {
+        this.table = table;
+        this.addColumnDefinitions = addColumnDefinitions;
+    }
+
+    public List<AddColumnDefinition<T>> tableElements() {
+        return addColumnDefinitions;
+    }
```
Comment (BaurzhanSakhariev) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java:39]: ditto

File: libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    public int hashCode() {
+        return Objects.hash(table, addColumnDefinitions);
+    }
+
+    @Override
+    public String toString() {
+        return "AlterTableAddColumn{" +
+               "table=" + table +
+               ", addColumnDefinition=" + addColumnDefinitions +
+               '}';
```
Comment (BaurzhanSakhariev) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java:68]: ditto

**Review by BaurzhanSakhariev — COMMENTED** [review]
Thanks! Left some minor comments, looks like just typos.

File: docs/sql/statements/alter-table.rst (hunk 1 above)
Comment (matriv) [reply docs/sql/statements/alter-table.rst:182]: thx, was more of starting a place holder to continue writing here once the whole feature is done.

File: libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(table, dropColumnDefinitions);
+    }
+
+    @Override
+    public String toString() {
+        return "AlterTableAddColumn{" +
+               "table=" + table +
```
Comment (BaurzhanSakhariev) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/AlterTableDropColumn.java:66]:
```suggestion
        return "AlterTableDropColumn{" +
               "table=" + table +
```

**Review by BaurzhanSakhariev — APPROVED** [review]
Spot some minor typo, otherwise LGTM

File: libs/sql-parser/src/main/java/io/crate/sql/tree/DropColumnDefinition.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, ifExists);
+    }
+
+    @Override
+    public String toString() {
+        return "DropAddColumnDefinition{name=" + name + ", ifExists=" + ifExists + '}';
+    }
```
Comment (BaurzhanSakhariev) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/DropColumnDefinition.java:66]:
```suggestion
    public String toString() {
        return "DropColumnDefinition{name=" + name + ", ifExists=" + ifExists + '}';
    }
```

**Comment by matriv** [comment]
> Spot some minor typos, otherwise LGTM

Many thx!

**Pull request merged**